values.yaml

```diff
@@ -77,6 +77,29 @@ global:
   # Additionally requires Consul >= 1.4 and consul-k8s >= 0.10.1.
   bootstrapACLs: false
 
+  # consulNamespaces are an Enterprise only feature available in Consul v1.7+.
+  # A valid Consul Enterprise license is required to make use of this feature.
+  # Requires consul-k8s v0.x+
+  consulNamespaces:
```

Comment:
I thought this was going under `syncCatalog`?

Reply:
What about the Connect injector then? It needs this same info. Are you suggesting we duplicate it both places?

Reply:
I thought that's what we had discussed on the RFC but I see now that when you updated it you only split up the allow/deny lists.

I'm trying to think through the use-cases:
1. Only using catalog sync - doesn't matter in global or syncCatalog
1. Only using connect inject - doesn't matter in global or connectInject
1. Using both catalog sync and connect inject - I can think of some use-cases where it might be nice to manage them separately: I'm using connect inject with namespace mirroring but have a couple of services that I can't put sidecars on and I want them still registered in Consul but all in the same Consul namespace called "k8s-sync".  Why? I don't know 😄.

I think on the tradeoff, we enable some use-cases if we split them up at the cost of having more config. I don't think it significantly complicates the chart though because in the end, the configs are localized to their own components: sync and inject. Where it might complicate things is in the acl bootstrap. I'm just looking at that PR now so don't have the context there yet. Would it make things a lot more difficult?

Reply:
I could see putting the Consul namespace under both the sync and injector. That should allow independent management of namespaces for those two processes. 

I don't think it makes sense to have the enable namespace flag in both though. Either you can run namespaces or not. It gets hard to reason about that value if it's turned on one place and off in another.

Reply:
> I don't think it makes sense to have the enable namespace flag in both though. Either you can run namespaces or not. It gets hard to reason about that value if it's turned on one place and off in another.

Yeah that makes sense as a global setting. That flag is really asking the user if they're running Consul Enterprise right? Maybe that's a better name? Or `namespacesEnabled` or something. Because it seems like it's something that the user can turn on but really they can't, it's more telling us whether they have that capability or not.

Reply:
Left the namespace enabling as a global flag, moved the destination namespace and mirroring options under both sync catalog and the connect injector.